### PR TITLE
Django 2.2

### PR DESCRIPTION
**Pull request by madhuBayy**
@ldko @somexpert 

This PR upgrades django -locking to Django 2.2. In this PR, I added a settings.py file to use `django-admin` commands to run the app and tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+License
+-------
+
+See LICENSE
+
+Contributors
+------------
+
+* [Gio Gottardi](https://github.com/somexpert)
+* [Madhulika Bayyavarapu](https://github.com/madhulika95b)
```
Comment (ldko) [inline README.md:91]: Since this is forked from debrouwere, we should list him too. It looks like he was the only contributor on that project.

File: setup.py [hunk 2]
```diff
@@ -1,27 +1,26 @@
 import os
 from setuptools import setup, find_packages
-version = '0.3.0'
+version = '1.0.0'
 README = os.path.join(os.path.dirname(__file__), 'README')
```
Comment (ldko) [inline setup.py]: The file name needs to be updated to 'README.md'.

File: setup.py [hunk 3]
```diff
@@ -1,27 +1,26 @@
...
 long_description = open(README).read()
 setup(name='locking',
       version=version,
+      packages=find_packages(),
+      install_requires=['django-staticfiles','simplejson'],
```
Comment (ldko) [inline setup.py]: Do we still need `django-staticfiles` here? It seemed to install one that is Python 2 version for me. Also, I don't think we need `simplejson` here.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+3. Install the requirements (preferably in a virtual environment):
+
+    ```python
+        python setup.py install
+    ```
+
+4. Run the migrations:
+
+    ```python
+        django-admin migrate
```
Comment (ldko) [inline README.md]: I installed Django 2.2 (should that be something added to `install_requires`?) then when I run this I get:
```
django.core.exceptions.ImproperlyConfigured: Requested setting DATABASES, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.
```
Is there an expectation certain things are already done to have these commands work?

**Review by ldko — CHANGES_REQUESTED** [review]
Based only what is in the README.md instructions, I didn't get this app running. Have you run through the steps there in a fresh environment to verify the steps work for you?

File: setup.py (hunk 3 above)
Comment (madhuBayy) [reply setup.py]: I'll check and update.

File: README.md (hunk 4 above)
Comment (madhuBayy) [reply README.md]: oh...yeah I had to set environment variables in my virtualenv before running these commands using - 

`export DJANGO_SETTINGS_MODULE=locking.settings`
`export PYTHONPATH="$(pwd)"`

Comment (madhuBayy) [reply README.md]: Do either of you think that we should have a `manage.py` and use manage.py commands to run and test the app similar to other apps?

Comment (ldko) [reply README.md]: That did cross my mind--I think it would be ok to add this.

File: setup.py (hunk 3 above)
Comment (madhuBayy) [reply setup.py]: I tried running this app, migrations and tests without these and it work just fine. I'll get rid of this from setup.py

File: locking/settings.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import os
```
Comment (ldko) [inline locking/settings.py]: I wonder if this settings file should be in the /tests directory.

File: README.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    ```python
+        python manage.py createsuperuser
+    ```
+
+6. Start the test server:
+
+    ```python
+        python manage.py runserver
+    ```
+
```
Comment (ldko) [inline README.md:71]: Should we add a step saying where to view the interface we are running?

File: setup.py [hunk 7]
```diff
@@ -1,27 +1,25 @@
...
                    'Topic :: Software Development :: Libraries :: Python Modules',
                    'Topic :: Utilities'],
-      keywords='locking mutex',
-      author='Stijn Debrouwere',
-      author_email='stijn@stdout.be',
-      url='http://stdbrouw.github.com/django-locking/',
-      download_url='http://www.github.com/stdbrouw/django-locking/tarball/master',
-      license='BSD',
-      packages=find_packages(),
-      install_requires=['django-staticfiles','simplejson'],
```
Comment (ldko) [inline setup.py:26]: I am wondering if we should add `django` as a dependency here in `install_requires`.

**Review by ldko — COMMENTED** [review]
With the latest changes I can successfully `runserver`, however, if I go to `/admin` the assets 404. If I try to add a `story` via that admin, I get a 500 error on save. Is this app working for you on its own @madhulika95b ? I know we don't usually run it independently, but it seems like it should work...

File: setup.py (hunk 7 above)
Comment (madhuBayy) [reply setup.py:26]: well...I think since in the README we are mentioning python and Django as system dependencies, it is expected to have them installed before using the app. would it be necessary to have it here?

File: README.md (hunk 6 above)
Comment (madhuBayy) [reply README.md:71]: yeah...I will add it here.

**Comment by madhuBayy** [comment]
> With the latest changes I can successfully `runserver`, however, if I go to `/admin` the assets 404. If I try to add a `story` via that admin, I get a 500 error on save. Is this app working for you on its own @madhulika95b ? I know we don't usually run it independently, but it seems like it should work...

Hmmm...when I ran this with `django-admin` command previously, I did run into 404 errors on some urls but could login to admin interface. But now I encountered the same error you mentioned with both `django-admin` and `python manage,py`. Let me look into this and fix it.

**EDIT**: By running `python manage.py migrate --run-syncdb`, the necessary tables are created, and stories can be added now fine.

File: setup.py (hunk 7 above)
Comment (ldko) [reply setup.py:26]: I wouldn't say it is necessary, but since we are instructing people to use a virtual environment, it would make sense to me they would install Django there and not as a system level dependency. We usually have django 2.2 in a test requirements file that we instruct people to install. Since we aren't using requirements files, I though tit might make sense to put it in the `install_requires`. What do you think @somexpert ?

Comment (somexpert) [reply setup.py:26]: I don't have strong feelings about this, since I don't think I've been very consistent with it myself... We _do_ mention virtual environments in the development section, but not for the main installation. In either case, having it in `install_requires` shouldn't be an issue, so I say we do it and get rid of it from the README.

**Comment by somexpert** [comment]
So I can run the tests and the test server and I can add Story objects, _but_ I run into issues when I lock them. I can lock/unlock them and view their locked status just fine by hitting the appropriate `/ajax/admin/tests/story/[record ID #]/[lock|unlock|is_locked]` endpoints, but in the normal admin interface, you can see that we're having trouble getting the staticfiles, which do exist in the app. So I think there's some disconnect somewhere in the test setup. To be clear, here's how to replicate the issue:
1. Start the app up as instructed in the README
2. Create a story object
3. Lock it by hitting `/ajax/admin/tests/story/1/lock`
4. See the missing icon and other missing resources by inspecting the requests when you hit `/admin/tests/story`, where it _should_ show some info on the lock status of the records.

File: README.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
+Django-locking
```
Comment (ldko) [inline README.md]: I would make this lowercase for consistency: django-locking.

File: README.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+        'locking',
+    )
+    ```
+
+Developing
+----------
+
+1. Clone the git repository:
+
+    ```python
```
Comment (ldko) [inline README.md]: It doesn't make much difference here based on the included text not triggering syntax coloring, but using ```console` makes more sense to me here, since these are commands run in the terminal.

File: README.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+        python manage.py createsuperuser
+    ```
+
+7. Start the test server:
+
+    ```python
+        python manage.py runserver
+    ```
+   The admin site of the test server can be viewed from a browser by navigating to the default location: http://localhost:8000/admin
+Testing
```
Comment (ldko) [inline README.md:72]: We need an empty line before this line; currently the README renders without a separate section for `Testing` otherwise.

**Review by ldko — COMMENTED** [review]
This is working for me how. I have a few small comments on the README.md. Thanks, @madhulika95b !

**Review by ldko — APPROVED** [review]
Works for me!

File: README.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+    ```console
+        python setup.py install
+    ```
+
+4. Run the migrations:
+
+    ```console
+        python manage.py migrate --run-syncdb
+    ```
+5. Collect all static files:
```
Comment (somexpert) [inline README.md]: Why was this added? Generally this is only needed for real deployments, not development, as the built-in Django server knows how to grab the static files when they're in predictable places, as well as it's own provided static files. I tried testing _without_ running this command and it seemed to work fine for me, so I believe we can remove this.

**Review by somexpert — APPROVED** [review]
This all works for me, but I think we can remove the `collectstatic` command. But it's not a big deal to me, I think it's alright either way, I just think it is unnecessary. :thumbsup:

File: README.md (hunk 11 above)
Comment (madhuBayy) [reply README.md]: Hmmm...when I was trying it on my system, the images weren't getting loaded and after running `collectstatic`, the images were getting loaded fine.

Comment (ldko) [reply README.md]: When I was reviewing the PR last week, I also tried it without running `collectstatic` and it seemed to be ok. Then I ran it after and it gave a lot of warnings about not copying files that already existed or something like that, but was also ok, so I didn't comment, but also at the time thought about asking if we needed it...

Comment (madhuBayy) [reply README.md]: ok, I'll try to run it in  a fresh virtual env and remove it if its unncessary. :)

Comment (madhuBayy) [reply README.md]: It worked for me too in a new virtualenv with a freshly cloned repo. I got rid of `collectstatic` from Readme. Thanks!!
